### PR TITLE
Document envvar COCOTB_LIBRARY_COVERAGE.

### LINKED ISSUE #2083: Document COCOTB_LIBRARY_COVERAGE envvar
The environment variable ``COCOTB_LIBRARY_COVERAGE`` -
https://github.com/cocotb/cocotb/blob/b2eda5325a1e75ef6bdde4e6523bcb0ac81a3137/cocotb/__init__.py#L160
is not documented.

### PR DESCRIPTION
**Pull request by cmarqu**
Closes #2083

### CODE REVIEW

**Comment by imphil** [comment]
Is that really a variable we want to document for end users? I see it more as an internal helper that might change at any time, and users should not rely on it (and shouldn't have any reason to use it, actually).

**Comment by eric-wieser** [comment]
It might still be nice to have it documented somewhere for cocotb maintainers, but I agree that it probably doesn't belong next to the user environment variables.

**Comment by imphil** [comment]
@cmarqu maybe you can find a more "cocotb developers-internal" section in the docs. If not, I'm fine with leaving it where it is now, but adding text like "For cocotb internal use only."

**Comment by cmarqu** [comment]
We already have documentation for cocotb-internal variables, and I figured consistency/completeness would be nice here.
I'll move these internals to a new section.

File: documentation/source/building.rst [hunk 1]
```diff
@@ -326,6 +315,27 @@ and
    `CXXFLAGS`, `LDLIBS` are not supported by distutils/pip
 
 
+Internal Variables
+------------------
+
+The following variables are used for cocotb-internals.
```
Comment (imphil) [inline documentation/source/building.rst:321]:
```suggestion
The following variables are used for cocotb internals.
```

**Review by imphil — APPROVED** [review]

**Review by ktbarrett — APPROVED** [review]

**Pull request merged**